lib/implore.es6.js

```diff
@@ -194,17 +194,16 @@ implore.delete = function httpDelete(options) {
 function getURLFromRequest(request) {
 	const queryString = makeQueryString(request.query || {});
 	let formatted = request.route;
-	let name;
 	let value;
 	let regexp;
 
-	for (name in request.params) {
-		if (request.params.hasOwnProperty(name)) {
-			value = request.params[name];
-			regexp = new RegExp(':' + name + '(?=(\\\/|$))');
-			formatted = formatted.replace(regexp, value);
-		}
-	}
+	// Replace route params with their values. If the route param is specified
+	// with an extension (e.g., ':id.json'), keep the extension.
+	Object.keys(request.params || {}).forEach((name) => {
```

Comment:
I'm not too sure about preference; we use the native `Object.keys(...).forEach` [throughout FluxThis](https://github.com/addthis/fluxthis/search?utf8=%E2%9C%93&q=forEach). But, in any case, I think the `each` helper doesn't fit as well here as the native alternative. The main benefit of the `each` helper is that you can break out of the loop early -- but that use case isn't relevant here. Beyond that, the native version has the benefit of being idiomatic JS, being slightly more performant (presumably, anyway, since it's native code and doesn't require an additional comparison operation on every loop), and having fewer dependencies (leaving a dependency on `each` out of `implore`, even if it *is* a dependency included with FluxThis). So, I'd like to leave it as is, unless I'm missing a strong case for doing it otherwise.

Also, I'm not sure about this (and it doesn't really matter anyway), but I *think* I might count as a FluxThis developer, even though I'm not one of the OGs. 😛